code/modules/projectiles/guns/projectile/CMturret.dm

```diff
@@ -0,0 +1,228 @@
+/mob/living/carbon/human
+	var/obj/structure/machinegun/mounted
+
+	ClickOn(var/atom/A, params)
```

Comment:
Replace this with
```
/mob/living/carbon/human/ClickOn(atom/A, params)
    if(mounted && Adjacent(mounted))
        if(next_move < world.time)
            mounted.shoot(get_turf(A))
    else
        . = ..()
```
then remove all the stupid nextshot shit and use user.changeNext_move(CLICK_CD_RANGE)